lib/Rex/Pkg/Debian.pm

```diff
@@ -32,7 +32,7 @@ sub new {
     install_version =>
       'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::=--force-confold --force-yes -y install %s=%s',
     update_system =>
-      'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq upgrade',
+      'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq update && APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq upgrade',
```

Comment:
On Sun, 2015-02-01 at 15:22 -0800, Ferenc Erki wrote:
> In lib/Rex/Pkg/Debian.pm:
> 
> > @@ -32,7 +32,7 @@ sub new {
> >      install_version =>
> >        'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::=--force-confold --force-yes -y install %s=%s',
> >      update_system =>
> > -      'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq upgrade',
> > +      'APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq update && APT_LISTCHANGES_FRONTEND=text DEBIAN_FRONTEND=noninteractive apt-get -y -qq upgrade',
> 
> What is the reason behind using a different command for repo update
> here than the simple update_package_db case?

Good point, calling update_pacakge_db would be better, but is that
possible with the current code design? It looks like you're defining a
system command to run, rather than a CODEREF. Should the hash be changed
to a hash of coderefs?

Andy